prowler/providers/aws/services/sns/sns_service.py

```diff
@@ -61,18 +61,25 @@ def _get_topic_attributes(self, regional_clients):
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
-    def _list_tags_for_resource(self):
-        logger.info("SNS - List Tags...")
+    def _list_tags_for_resource(self, resource: any):
+        logger.info("SNS - Listing Tags...")
         try:
-            for topic in self.topics:
-                regional_client = self.regional_clients[topic.region]
-                response = regional_client.list_tags_for_resource(
-                    ResourceArn=topic.arn
-                )["Tags"]
-                topic.tags = response
+            if getattr(resource, "region", "") and getattr(resource, "arn", ""):
```

Comment:
Why this if you are expecting a `Topic`?

Reply:
In the function's parameters, I defined the resource as `any` in case we want to list tags for another resource in the future. However, I believe this is more theoretical than practical. Although I defined it as `any` we control when the function is called, and it's unlikely we'd call something that doesn't have an ARN. I can remove it if you think it's unnecessary.

Reply:
I think for now is more descriptive if we just use the `Topic` with their attributes, although I appreciate having that generic point of doing things :D